Source/AwsCommonRuntimeKit/mqtt/MqttRequestResponseClient.swift

```diff
@@ -60,17 +51,36 @@ public struct SubscriptionStatusEvent: Sendable {
     public let error: CRTError?
 }
 
-// TODO: Igor has updated the events for IoT Command. Need update later
 /// An event that describes an incoming publish message received on a streaming operation.
 public struct IncomingPublishEvent: Sendable {
 
+    /// The topic associated with this PUBLISH packet.
+    public let topic: String
+    
     /// The payload of the publish message in a byte buffer format
-    let payload: Data
+    public let payload: Data
+    
+    /// (Optional) The content type of the payload
+    public let contentType: String?
 
-    /// The topic associated with this PUBLISH packet.
-    let topic: String
+    /// (Optional) User Properties, if there is no user property, the array is empty
```

Comment:
Trivial: Should we allow this to be nil by making it optional ( [UserProperty]? ) ? Other optional properties are treated as optional so it feels a little off for this "Optional" to not be.